docs/how-to/cli-apps.rst

```diff
@@ -0,0 +1,157 @@
+------------------------------
+
+Briefcase provides two ways to run your app: ``dev`` mode and ``run`` mode.
+You can pass command-line arguments to your app in both cases using ``--`` followed
+by your arguments.
+
+For example, running in dev mode:
+
+.. code-block:: console
+
+    $ briefcase dev -- John
+    ===========================================================================
+    Hello, John!
+
+And similarly after packaging:
+
+.. code-block:: console
+
+    $ briefcase run -- John
+    ===========================================================================
+    Hello, John!
+
+This is a key difference from GUI apps, which are usually launched without passing
+arguments and display a graphical window.
+
+Handling Command-line Arguments
+-------------------------------
+
+You can use any standard Python tool to parse CLI arguments. Here's a simple example
+using :any:`argparse`:
+
+.. code-block:: python
+
+    import argparse
+
+    def main():
+        parser = argparse.ArgumentParser(description="A simple CLI app.")
+        parser.add_argument("name", help="Your name")
+        args = parser.parse_args()
+
+        print(f"Hello, {args.name}!")
+
+Update your ``app.py`` file with this logic. You can now pass a name when running the app:
+
+.. code-block:: console
+
+    $ briefcase dev -- John
+    ===========================================================================
+    Hello, John!
+
+Packaging and Distribution
+---------------------------
+
+Once you've tested your app, you can package it for distribution using:
+
+.. code-block:: console
+
+    $ briefcase create
+    $ briefcase build
+    $ briefcase package
+
+For **macOS**, console apps are packaged as ``.pkg`` files, rather than the ``.app``
+bundles used for GUI apps. This is another major distinction.
+
+.. code-block:: console
+
+    $ briefcase package --adhoc-sign
+    ...
+    [hello-cli] Packaged dist/Hello CLI-0.0.1.pkg
+
+This ``.pkg`` file installs the app globally, and it can run from the terminal:
+
+.. code-block:: console
+
+    $ hello-cli John
+    Hello, John!
+
+On **Linux**, you'll get a ``.deb`` package, and on **Windows**, a ``.msi`` installer.
+
+Console App Behavior Summary
+----------------------------
+
+Key differences from GUI apps include:
+
+- **Project structure** includes ``console_app = true`` in ``pyproject.toml``
+- **Apps are executed in the terminal**, and print directly to stdout
+- **Arguments are passed using** ``-- ARGS ...`` when using ``briefcase dev`` and ``briefcase run``
+- **No GUI framework or windowing system** is used or bundled
+- **macOS packaging uses ``.pkg`` format**, rather than GUI `.app` bundles
+
+What's next?
+------------
+
+Now that you've built a simple console app with Briefcase, you can use these
+principles to build more complex CLI tools or integrate other libraries like
+`Click <https://click.palletsprojects.com/>`_, `Typer <https://typer.tiangolo.com/>`_,
+or any other Python CLI framework.
```

Comment:
It might also be worth mentioning curses and Textual and here, as a "GUI but in text mode" options.